### ISSUE TITLE
h2o exits when reverse proxying: HTTP/2 only

### ISSUE THREAD

**Issue by Lukasa** [h2o/h2o#185]
I'm trying to run h2o as a reverse proxy for my Python webapp. This is working fine when h2o receives HTTP/1.1 requests, but when it receives HTTP/2 requests it encounters a fatal error and exits. The specific fatal error it hits is [this one](https://github.com/h2o/h2o/blob/1985681112d8dc8683d7d361164263abf835a564/lib/http2/hpack.c#L470).

This error seems to be to do with the HPACK encoding for the response. I can trigger this error when I use Chrome as a client, and when I use `hyper` (my own HTTP/2 implementation) as a client.

The header causing the problem seems to be `access-control-allow-origin: *`. The act of encoding the * seems to lead h2o into trouble.

A GDB backtrace is below:

```
fatal:value out of range

Program received signal SIGABRT, Aborted.
0x00007ffff6fd4cc9 in __GI_raise (sig=sig@entry=6)
    at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
56	../nptl/sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) backtrace
#0  0x00007ffff6fd4cc9 in __GI_raise (sig=sig@entry=6)
    at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007ffff6fd80d8 in __GI_abort () at abort.c:89
#2  0x0000000000407874 in h2o_fatal (
    msg=msg@entry=0x42ba9a "value out of range")
    at /h2o-1.0.0/lib/common/memory.c:60
#3  0x000000000041c403 in encode_int (dst=dst@entry=0x6c927d "\200", 
    value=<optimized out>, prefix_bits=prefix_bits@entry=7)
    at /h2o-1.0.0/lib/http2/hpack.c:469
#4  0x000000000041cf9d in h2o_hpack_encode_string (
    _dst=_dst@entry=0x6c927d "\200", s=0x6aa320 "*", len=1)
    at /h2o-1.0.0/lib/http2/hpack.c:521
#5  0x000000000041d197 in encode_header (
    header_table=header_table@entry=0x6c79c0, dst=0x6c927d "\200", 
    name=0x63a670 <h2o.tokens+240>, value=value@entry=0x6aa2e0)
    at /h2o-1.0.0/lib/http2/hpack.c:585
#6  0x000000000041d576 in flatten_headers (num_headers=2, 
    headers=<optimized out>, header_table=0x6c79c0, dst=<optimized out>)
    at /h2o-1.0.0/lib/http2/hpack.c:636
#7  h2o_hpack_flatten_response (buf=buf@entry=0x6c7a30, 
    header_table=header_table@entry=0x6c79c0, stream_id=1, 
    max_frame_size=16384, res=res@entry=0x6c7da8, ts=ts@entry=0x7fffffffe1f0, 
    server_name=0x676758 <conf+24>) at /h2o-1.0.0/lib/http2/hpack.c:723
#8  0x000000000041df88 in send_headers (conn=conn@entry=0x6c78d0, 
    stream=stream@entry=0x6c7c10) at /h2o-1.0.0/lib/http2/stream.c:206
#9  0x000000000041e120 in finalostream_send (self=0x6c7c18, req=0x6c7d00, 
    bufs=0x7fffffffe280, bufcnt=1, is_final=1)
    at /h2o-1.0.0/lib/http2/stream.c:253
#10 0x00000000004150f8 in do_send (self=<optimized out>)
    at /h2o-1.0.0/lib/handler/proxy.c:195
#11 0x0000000000415199 in on_body (client=<optimized out>, 
    errstr=<optimized out>) at /h2o-1.0.0/lib/handler/proxy.c:223
#12 0x000000000040882d in on_body_content_length (sock=0x6c6600, status=0)
    at /h2o-1.0.0/lib/common/http1client.c:112
#13 0x000000000040aa63 in read_on_ready (sock=0x6c6600)
    at /h2o-1.0.0/lib/common/socket/evloop.c.h:228
#14 run_socket (sock=0x6c6600) at /h2o-1.0.0/lib/common/socket/evloop.c.h:466
#15 run_pending (loop=<optimized out>)
    at /h2o-1.0.0/lib/common/socket/evloop.c.h:479
#16 h2o_evloop_run (loop=0x6ad110)
    at /h2o-1.0.0/lib/common/socket/evloop.c.h:492
#17 0x000000000041f4a2 in run_loop (_thread_index=_thread_index@entry=0x0)
    at /h2o-1.0.0/src/main.c:1106
#18 0x0000000000406afb in main (argc=<optimized out>, argv=<optimized out>)
    at /h2o-1.0.0/src/main.c:1334
```

**Comment by kazuho** [comment]
Thank you for reporting the issue.  The problem has been fixed and merged to master in #186.

Please reopen if the problem still persists.

**Comment by Lukasa** [comment]
Thanks! :cake:
